### ISSUE TITLE
Improve exception raised at PubControlClient#pubcall

### ISSUE THREAD

**Issue by luax** [fanout/ruby-pubcontrol#4]
Hi,

It would be great if the exception raised at (https://github.com/fanout/ruby-pubcontrol/blob/master/lib/pubcontrolclient.rb#L145) was improved to contain more data:
```
    if !response.kind_of? Net::HTTPSuccess
      raise 'failed to publish: ' + response.class.to_s + ' ' +
          response.message
    end
```

Today if the publish fails with a bad request, an exception is raised with the message: 

```
failed to publish: Net::HTTPBadRequest Bad Request
```

But this does not tell us any root cause, perhaps there is a body attached to the bad request response that has some explanation to it? And it's not possible to extract the response.

I know that error handling and Net is hard but it would be nice to be able to catch the exception with the response data etc

Thank you!

**Comment by jkarneges** [comment]
Published version 1.2.3 which includes the response body in the raised exception.

**Comment by luax** [comment]
Great, thanks! @jkarneges
